### DISCUSSION TITLE
Inheritance of filter

### DISCUSSION THREAD

**Discussion by ravensnowbird** [AaronLasseigne/active_interaction#550] [Ideas]
When i use this in an API - I have a parent `ApplicationInteractor`

```
class ApplicationInteractor < ActiveInteraction::Base
  hash :params do
    string :user_email
    string :authentication_token
    integer :company_id
  end
  
  validate :current_user
  def current_user
     ... authenticate user ..
  end
end
```

```
class CreateFooInteractor < ApplicationInteractor
  hash :params do
    integer :id
    string :name
  end
  
  def execute
     ... create foo ...
  end
end
```
hash params is getting overridden in `CreateFooInteractor` and its forcing me to use `strip: false` and I feel it would be nice to have this extended so I have some Interactions using filters

**Comment by AaronLasseigne** [comment]
>  I feel it would be nice to have this extended so I have some Interactions using filters

I'm not understanding what you want this to do. Can you provide an example of how you would like it to work?
